File: Dec2020/19Dec/CountTheNumberOfConsistentStrings.cpp
/*
Problem name: 1684. Count the Number of Consistent Strings
Problem link: https://leetcode.com/problems/count-the-number-of-consistent-strings/
*/

class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        set<char> a;
        for(int i=0;i<allowed.length();++i)
            a.insert(allowed[i]);
        
        int count=0, flag;
        for(int i=0;i<words.size();++i){
            flag=0;
            for(int j=0;j<words[i].size();++j){
                if(!a.count(words[i][j])){
                    flag=1;
                    break;
                }
            }
            if(!flag)  count++;
        }
        return count;
    }
};

File: Dec2020/19Dec/CountTheNumberOfConsistentStrings.java
/*
Problem Name: 1684. Count the Number of Consistent Strings
Problem Link: https://leetcode.com/problems/count-the-number-of-consistent-strings/
*/
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        int allow[] = new int[26];
        for(char c:allowed.toCharArray())
            allow[c - 'a'] = 1;
        int consistent_count = 0;
        for(int i=0;i<words.length;i++){
            int word[] = new int[26];
            boolean consistent = true;
            for(char c:words[i].toCharArray())
                if(allow[c - 'a'] != 1){
                    consistent = false;
                    break;
                }
            if(consistent)
                consistent_count++;
        }
        return consistent_count;
    }
}
